pkg/scaler/prober.go

```diff
@@ -47,7 +47,8 @@ const (
 	defaultJitterMaxFactor     = 0.2
 	defaultJitterSliding       = true
 
-	kindDeployment = "Deployment"
+	kindDeployment             = "Deployment"
+	ignoreScalingAnnotationKey = "gardener.cloud/ignore-scaling"
```

Comment:
Although I checked the annotation is not used anywhere in gardener org code base, however I was wondering if it is too generic to differentiate it's usage. 
Shall we add something to distinguish it from scaling operation done via other operations.

Reply:
how about `test.gardener.cloud/ignore-scaling-during-IT`

Reply:
Having a test annotation is not such a great idea because now DWD needs to react to test annotations in its code. There are therefore 3 approaches: 

1. We can follow a similar approach to what a `gardener-resource-manager` does. It looks for  `resources.gardener.cloud/ignore=true` annotation on `ManagedResources` and if any such annotation is present then that resource will not be reconciled.
2. As you suggested, add an annotation to `DWD` to ignore scaling via annotation `gardener.cloud/ignore-scaling`
3. Update the configmap and add an additional property against `CA` to ignore scaling.

Let us analyze the approaches a bit:

**Approach-1:** 
This would mean that `cluster-autoscaler` resource will now have an additional annotation. Assumption is that the only component that will react to this annotation will be `DWD`. To ensure that change in the annotation is reflected in its informer cache, it could use the `initial-delay` field in the configuration for any scaling (scaling up or scaling down) operation. 

**Approach-2:** 
Adding an annotation `gardener.cloud/ignore-scaling` is ambiguous. As this annotation does not reflect on scaling of which resource should be avoided. Ideally if you only wish to prevent scaling of `CA` then you should be using a more specific annotation - example:  `resources.gardener.cloud/cluster-autoscaler/ignore-scaling`. 

**Approach-3:** 
Dynamically change the config map that is used by `DWD` to add additional field `ignore-scaling` with value `true` or any truthy value. Since this is a namespaced resource it is also an acceptable solution but it will involve more work, since you change the mounted config-map, DWD now needs to be capable of auto-reload the configuration and ensure that all probes that are started also are using the newly parsed configuration. This is a LOT of work.